content/blog/2017/2017-01-19-converting-conditional-to-pipeline.adoc

```diff
@@ -0,0 +1,512 @@
+        booleanParam (
+            defaultValue: false,
+            description: '',
+            name : 'FORCE_FULL_BUILD')
+    }
+
+    stages {
+        stage ('Prepare') {
+            steps {
+                checkout([$class: 'GitSCM',
+                    branches: [[name: "origin/${BRANCH_PATTERN}"]],
+                    doGenerateSubmoduleConfigurations: false,
+                    extensions: [[$class: 'LocalBranch']],
+                    submoduleCfg: [],
+                    userRemoteConfigs: [[
+                        credentialsId: 'bitwiseman_github',
+                        url: 'https://github.com/bitwiseman/hermann']]])
+            }
+        }
+
+        stage ('Build') {
+            when {
+                expression {
+                    GIT_BRANCH = 'origin/' + sh(returnStdout: true, script: 'git rev-parse --abbrev-ref HEAD').trim()
+                    return GIT_BRANCH == 'origin/master' || params.FORCE_FULL_BUILD
+                }
+            }
+            steps {
+                // Freestyle build trigger calls a list of jobs
+                // Pipeline build() step only calls one job
+                // To run all three jobs in parallel, we use "parallel" step
+                // https://jenkins.io/doc/pipeline/examples/#jobs-in-parallel
+                parallel (
+                    linux: {
+                        build job: 'full-build-linux', parameters: [string(name: 'GIT_BRANCH_NAME', value: GIT_BRANCH)]
+                    },
+                    mac: {
+                        build job: 'full-build-mac', parameters: [string(name: 'GIT_BRANCH_NAME', value: GIT_BRANCH)]
+                    },
+                    windows: {
+                        build job: 'full-build-windows', parameters: [string(name: 'GIT_BRANCH_NAME', value: GIT_BRANCH)]
+                    },
+                    failFast: false)
+            }
+        }
+        stage ('Build Skipped') {
+            when {
+                expression {
+                    GIT_BRANCH = 'origin/' + sh(returnStdout: true, script: 'git rev-parse --abbrev-ref HEAD').trim()
+                    return !(GIT_BRANCH == 'origin/master' || params.FORCE_FULL_BUILD)
+                }
+            }
+            steps {
+                echo 'Skipped full build.'
+            }
+        }
+    }
+}
+----
+
+
+== Conclusion
+
+As I said before, the Conditional BuildStep plugin is great.
+It provides a clear, easy to understand way to add conditional logic to any Freestyle job.
+Before Pipeline, it was one of the few plugins to do this and it remains one of the most popular plugins.
+Now that we have Pipeline, we can implement conditional logic directly in code.
+
+This is blog post discussed how to approach converting conditional build steps to Pipeline
+and showed a couple concrete examples.  Overall, I'm pleased with the results so far.
+I found scenarios which could not easily be migrated to Pipeline, but even those
```

Comment:
I seriously don't understand the point of this blog post.

"Look, there's a new way to do things and it's just as cumbersome as the old one, but different!"